edu_edfi_airflow/dags/earthbeam_dag.py

```diff
@@ -372,27 +391,56 @@ def build_tenant_year_taskgroup(self,
+                #     '{{ ds_nodash }}', '{{ ts_nodash }}',
+                #     "lightbeam_results.json"
+                # )
+                
                 lb_results_file = edfi_api_client.url_join(
                     self.emlb_results_directory,
                     tenant_code, self.run_type, api_year, grain_update,
                     '{{ ds_nodash }}', '{{ ts_nodash }}',
                     "lightbeam_results.json"
                 )
+                
+                if self.dag.params['send']:
+                
+                    send_lightbeam = LightbeamOperator(
+                        task_id=f"{taskgroup_grain}_send_via_lightbeam",
+                        lightbeam_path=self.lightbeam_path,
+                        data_dir=airflow_util.xcom_pull_template(run_earthmover.task_id),
+                        state_dir=lb_state_dir,
+                        results_file=lb_results_file if logging_table else None,
+                        edfi_conn_id=edfi_conn_id,
+                        **(lightbeam_kwargs or {}),
+                        pool=self.lightbeam_pool,
+                        command='send',
+                        dag=self.dag
+                    )
 
-                run_lightbeam = LightbeamOperator(
-                    task_id=f"{taskgroup_grain}_send_via_lightbeam",
-                    lightbeam_path=self.lightbeam_path,
-                    data_dir=airflow_util.xcom_pull_template(run_earthmover.task_id),
-                    state_dir=lb_state_dir,
-                    results_file=lb_results_file if logging_table else None,
-                    edfi_conn_id=edfi_conn_id,
-                    **(lightbeam_kwargs or {}),
-                    pool=self.lightbeam_pool,
-                    dag=self.dag
-                )
-
-                task_order.append(run_lightbeam)
-
+                    task_order.append(send_lightbeam)
+                
+                if self.dag.params['validate']:
+
+                    validate_lightbeam = LightbeamOperator(
+                        task_id=f"{taskgroup_grain}_validate_via_lightbeam",
+                        lightbeam_path=self.lightbeam_path,
+                        data_dir=airflow_util.xcom_pull_template(run_earthmover.task_id),
+                        state_dir=lb_state_dir,
+                        results_file=lb_validate_results_file if logging_table else None,
+                        edfi_conn_id=edfi_conn_id,
+                        validate_methods=self.dag.params['validate_methods'],
+                        validate_reference_fail_fast=self.dag.params['validate_reference_fail_fast'],
+                        **(lightbeam_kwargs or {}),
+                        pool=self.lightbeam_pool,
+                        command='validate',
+                        dag=self.dag
+                    )
+                    
+                    task_order.append(validate_lightbeam)
```

Comment:
Two qs on this section:
- do we expect cases where we'll want to have tasks for _both_ validate and send? Send does a validate within, right?
- if yes^, shouldn't validate come before send in the DAG?

Reply:
@AngelicaLastra @tomreitz